lib/orbit_web/auth/auth.ex

```diff
@@ -0,0 +1,56 @@
+defmodule OrbitWeb.Auth.Auth do
+  alias Orbit.Authentication.User
+  alias Orbit.Repo
+  alias Plug.Conn
+
+  @spec login(Conn.t(), String.t(), integer(), String.t() | nil, [String.t()], String.t() | nil) ::
+          Conn.t()
+  def login(conn, username, ttl_seconds, _refresh_token, _groups, logout_url) do
+    email = String.downcase(username)
+
+    Repo.transaction(fn ->
```

Comment:
**question:** what's the purpose of the transaction here? If it's to avoid having a record inserted with the same username between the `Repo.get_by/2` and the `Repo.insert!/1`, I don't believe the standard isolation level of Postgres transactions will accomplish that.

Reply:
Yeah, I think you're right. This is also in Glides, so I'll remove it here and then there in a separate PR 👍

Reply:
Thanks for making the change here and in Glides! It does still leave the question of how to best handle the race condition, but it's not likely to happen very often at all (it would require someone logging into the app for the first time roughly simultaneously from two devices or browsers), and if it ever does happen someone would just have to refresh and then the next request should work correctly, so I'm not too worried about it.